packages/lodestar/src/eth1/impl/ethers.ts

```diff
@@ -70,40 +70,63 @@ export class EthersEth1Notifier extends (EventEmitter as { new(): Eth1EventEmitt
       this.logger.verbose("Eth1 notifier is disabled" );
       return;
     }
-    if (this.started) {
-      this.logger.verbose("Eth1 notifier already started" );
+    await this.doStart();
+  }
+
+  public async stop(): Promise<void> {
+    this.provider.removeAllListeners("block");
+    // stop processing eth1
+    this.startedProcessEth1 = false;
+    this.logger.verbose("Eth1 notifier stopped");
+  }
+
+  /**
+   * This is triggered when building genesis or after chain gets started.
+   * @param subscribe true for genesis builder
+   * @returns a pushable for genesis builder, null otherwise.
+   */
+  public async startProcessEth1Blocks(subscribe = false): Promise<Pushable<[IDepositEvent[], ethers.providers.Block]>> {
+    if (this.startedProcessEth1) {
+      this.logger.info("Started processing eth1 blocks already");
       return;
     }
-    this.started = true;
-    if(!this.contract) {
-      await this.initContract();
+    if (!this.opts.enabled) {
+      if (subscribe) {
+        this.logger.info("Eth1 notifider is disabled but starting it to build genesis state");
+        await this.doStart();
+      } else {
+        this.logger.warn("Eth1 notifider is disabled, no need to process eth1 for proposing data");
+        return;
+      }
     }
-    const lastProcessedBlockTag = await this.getLastProcessedBlockTag();
-    this.lastProcessedEth1BlockNumber = (await this.getBlock(lastProcessedBlockTag)).number;
-    this.logger.info(
-      `Started listening to eth1 provider ${this.opts.provider.url} on chain ${this.opts.provider.network}`
-    );
-    this.logger.verbose(
-      `Last processed block number: ${this.lastProcessedEth1BlockNumber}`
-    );
     const headBlockNumber = await this.provider.getBlockNumber();
     // process historical unprocessed blocks up to curent head
     // then start listening for incoming blocks
+    if (subscribe) {
+      this.eth1Source = pushable<[IDepositEvent[], ethers.providers.Block]>();
```

Comment:
It's gonna bi nicer to use if it's object {depositEvents: IDepositEvent, block: Block}